src/settings.ts

```diff
@@ -136,6 +136,15 @@ export namespace Settings {
         label: i18n.__("settings.createSubfolder.label"),
         description: i18n.__("settings.createSubfolder.description"),
       },
+      createSubfolderPerProfile: {
+        value: false,
```

Comment:
I think we should enable this by default ... But not for already installed versions. 
Can be made in the function `upgradeBackupPluginVersion`.

Or what do you think?

Reply:
> Or what do you think?

Currently, [the upstream pull request](https://github.com/laurent22/joplin/pull/9942/files#diff-3cffaa2da8e9f556fa0211d50e67ce6ddcb1fc03743477c1b052bf27ebe7aae1R9) makes Joplin responsible for setting the default backup directory to the home directory (rather this plugin). As such, there are tradeoffs to having this setting enabled by default.

**Some reason(s) to not enable this by default**:
- **Extra directories**: If the default backup directory for this plugin is kept as the user's profile directory, backups would then look like this: `profileDir/JoplinBackup/default`. Because each subprofile has its own profile directory, subprofile backups would be in directories like this: `profileDir/profile-abcd/JoplinBackup/profile-abcd`. Observe that each `JoplinBackup` directory would have exactly one subdirectory.


**Some reasons to enable this by default**:
- **Consistency**: It will need to be enabled by default in the built-in version of Simple Backup to prevent backups from different profiles from overwriting one another. Enabling it here would then be consistent with the built-in version.
    - The <u>default</u> values of settings in built-in plugins can be [overridden here](https://github.com/laurent22/joplin/blob/f37866219676778c59c84b86bed78a0d34ff0f42/packages/lib/services/plugins/defaultPlugins/desktopDefaultPluginsInfo.ts#L9).
- **Data safety**: If a user manually changes multiple profiles to back up to the same location, having this setting enabled could prevent backups from being overwritten.

Reply:
To summarize, I think enabling this by default only makes sense in the version built-in to Joplin. In other versions, the default backup location is within the profile directory, which is different for each profile.